### ISSUE TITLE
WARN Unable to load properties from '/usr/local/UniFi/data/system.properties'

### ISSUE THREAD

**Issue by phoskin** [unofficial-unifi/unifi-pfsense#230]
Hi,

Ran the script on a new build on a jail on a FreeNAS 12.2 machine.

The service fails to start:
```
root@unifi:/usr/ports # service unifi.sh start
Starting UniFi controller.
root@unifi:/usr/ports # WARN Unable to load properties from '/usr/local/UniFi/data/system.properties' - /usr/local/UniFi/data/system.properties (No such file or directory)
Exception in thread "main" java.lang.NoClassDefFoundError: org/xerial/snappy/OSInfo
        at com.ubnt.service.system.S.<clinit>(Unknown Source)
        at com.ubnt.ace.Launcher.<clinit>(Unknown Source)
Caused by: java.lang.ClassNotFoundException: org.xerial.snappy.OSInfo
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:583)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
        ... 2 more
```

I tried to resolve that issue with mkdir /usr/local/UniFi/data;touch /usr/local/UniFi/data/system.properties - and was faced with this:
root@unifi:/usr/local/UniFi/data # service unifi.sh start
Starting UniFi controller.
```
root@unifi:/usr/local/UniFi/data # Exception in thread "main" java.lang.NoClassDefFoundError: org/xerial/snappy/OSInfo
        at com.ubnt.service.system.S.<clinit>(Unknown Source)
        at com.ubnt.ace.Launcher.<clinit>(Unknown Source)
Caused by: java.lang.ClassNotFoundException: org.xerial.snappy.OSInfo
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:583)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
        ... 2 more
```

**Comment by sikysikov** [comment]
Hi, I had similar issues, just follow Troubleshooting guide on main page. You have to uninstall java packages and reinstall UNIFI.

**Comment by Svedrin** [comment]
To expand on that comment a bit, I ran into this issue after installing UniFi on a Raspberry Pi. First it wouldn't even try to start and complain that it `Cannot locate Java Home`, which I fixed [by installing the `openjdk-8-jre-headless` package](https://neilen.com.au/2019/03/15/unifi-controller-cannot-local-java-home/). Then I ran into the issue documented here.

I couldn't find the Troubleshooting guide that the post above is referring to, but I did succeed by running `apt-get install --reinstall unifi` which seemed to generate the `system.properties` file correctly. After that, the Controller started up fine.
